extensions/2.0/Vendor/MSFT_CollisionPrimitives/README.md

```diff
@@ -0,0 +1,75 @@
+
+## glTF Schema Updates
+
+The `MSFT_CollisionPrimitives` extension can be added to any `node` to define one or more of the following properties:
+
+| |Type|Description|
+|-|-|-|
+|**collider**|`object`|Create a collision shape in the physics engine for this node.|
+
+### Colliders
+
+This extension provides a set of document-level objects, which can be referenced by nodes in the scene. The precise usage of these collider primitives should be specified by the extenions which utilizes the colliders. In general, these colliders are used to specify geometry which can be used for collision detection. Typically, the geometry specified by the collider will be simpler than any meshes used by the node or it's children, enabling real-time applications to perform queries such as intersection tests.
+
+Implementations of this extension should ensure that collider transforms are always kept in sync with node transforms - for example animated node transforms should be applied to the applications' internal representation of the collision geometry.
+
+To describe the shape that represents this node, should use, colliders must define exactly one of the following properties:
+
+| |Type|Description|
+|-|-|-|
+|**sphere**|`object`|A sphere centered at the origin in local space.|
+|**box**|`object`|An axis-aligned box centered at the origin in local space.|
+|**capsule**|`object`|A capsule centered at the origin and aligned along the Y axis in local space.|
+|**cylinder**|`object`|A cylinder centered at the origin and aligned along the Y axis in local space.|
+|**convex**|`object`|A convex hull wrapping a `mesh` object.|
+|**trimesh**|`object`|A triangulated representation of a `mesh` object.|
+
+The sphere, box, capsule, cylinder and convex types should all produce _solid_ colliders by default - a rigid body anywhere inside the shape should be pushed out of it.
+However the trimesh type always represents an infinitely thin shell or sheet - for example a trimesh collider created from a `mesh` object in the shape of a box will be represented as a hollow box.
+
+If you want your collider to have an offset from the local space (for example a sphere _not_ centered at local origin, or a rotated box), you should add an extra node to the hierarchy and apply your transform and your collider properties to that.
+
+**Collision Filtering**
+
+By default each `collider` can generate collisions with every other `collider`, provided they are sufficiently close. If you want certain objects in your scene to ignore collisions with others, you can provide the following optional collider properties:
+
+| |Type|Description|
+|-|-|-|
+|**collisionSystems**|`[string]`|An array of arbitrary strings indicating the `system` a node is in.|
```

Comment:
> An array of arbitrary strings indicating the `system` a node is in.

Labeling nodes with arbitrary strings (henceforth called "categories") could be very useful in other contexts. For example, `lightingCategories` in a punctual light could selectively light certain nodes or `renderCategories` in a camera could selectively render certain nodes. Should we consider breaking this labeling into a separate `node_categories` extension so other extensions have a standard way to reference all nodes in a set?

```json
{
  "nodes": [{
    "extensions": {
      "KHR_node_categories": {
        "categories": {
          "include": [0, 2]
        }
      }
    }
  }, {
    "extensions": {
      "KHR_node_categories": {
        "categories": {
          "exclude": [1]
        }
      }
    }
  }],
  "extensions": {
    "KHR_node_categories": {
      "categories": ["a", "b", "c", "d"]
    }
  }
}
```

A physics collider would only specify the `collisionCategories` because the node already defines its `categories`.

```json
{
  "collisionCategories": {
    "include": [0]
  },
  "box": {
    "size": [0.1, 0.2, 0.3]
  }
}
```